src/main/rascal/refactor/Rename.rsc

```diff
@@ -29,200 +29,200 @@ module refactor::Rename
 
 import refactor::TextEdits;
 
-import Message;
-import util::Reflective;
+import analysis::typepal::FailMessage;
+import analysis::typepal::Messenger;
+import analysis::typepal::TModel;
+
 
 import IO;
 import List;
+import Map;
+import Message;
 import Node;
+import ParseTree;
+import Set;
 
-data TModel;
-data Tree;
+import util::Reflective;
 
-data RenameState;
+alias RenameResult = tuple[list[DocumentEdit], set[Message]];
 
-alias RenameResult = tuple[list[DocumentEdit], map[str, ChangeAnnotation], set[Message]];
+data Renamer
+    = renamer(
+        void(FailMessage) msg
+      , void(DocumentEdit) documentEdit
+      , void(TextEdit) textEdit
+      , RenameConfig() getConfig
 
-data RenameSolver(
-        RenameResult() run = RenameResult() { throw "Not implemented"; }
-      , void(loc l, void(RenameState, Tree, RenameSolver) doWork, RenameState state) collectParseTree = void(loc _, void(RenameState, Tree, RenameSolver) _, RenameState _) { throw "Not implemented!"; }
-      , void(loc l, void(RenameState, TModel, RenameSolver) doWork, RenameState state) collectTModel = void(loc _, void(RenameState, TModel, RenameSolver) _, RenameState _) { throw "Not implemented!"; }
-      , void(Message) msg = void(Message _) { throw "Not implemented"; }
-      , void(DocumentEdit) documentEdit = void(DocumentEdit _) { throw "Not implemented"; }
-      , void(TextEdit) textEdit = void(TextEdit _) { throw "Not implemented"; }
-      , void(str, ChangeAnnotation) annotation = void(str _, ChangeAnnotation _) { throw "Not implemented"; }
-      , value(str) readStore = value(str _) { throw "Not implemented"; }
-      , void(str, value) writeStore = void(str _, value _) { throw "Not implemented"; }
-) = rsolver();
+      // Helpers
+      , void(str, loc) warning
```

Comment:
shouldn't these interfaces be extended to match the `FailMessage` possibilities?

Reply:
I swapped the arguments and now use `value` for the location, but unfortunately, Rascal does not syntactically allow the use of varargs here.